src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,227 @@
+package duke.command;
+
+import duke.exceptions.DukeIllegalDescriptionException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import java.util.Date;
+
+/**
+ * This is a class to make sense of the user commands and allow Duke to take different actions
+ * depending on the commands received.
+ */
+class Parser {
+
+    private String[] action;
+
+    /**
+     * Constructor of the Parser class.
+     *
+     * @param action A string array contains split commands.
+     */
+    Parser(String[] action) {
+        this.action = action;
+    }
+
+    /**
+     * To make amendments according to the first index of array which is the command type.
+     *
+     * @throws DukeIllegalDescriptionException To deal with the illegal and invalid inputs from users.
+     */
+    void parse() throws DukeIllegalDescriptionException {
+        switch (Command.valueOf(action[0])) {
+        case bye:
+            commandBye();
+            break;
+        case list:
+            commandList();
+            break;
+        case done:
+            commandDone();
+            break;
+        case todo:
+            commandTodo();
+            break;
+        case deadline:
+            commandDeadline();
+            break;
+        case event:
+            commandEvent();
+            break;
+        case delete:
+            commandDelete();
+            break;
+        case find:
+            commandFind();
+            break;
+        }
+
+    }
+
+    /**
+     * Action taken when the command is bye.
+     */
+    private void commandBye() {
+        System.out.println("Bye. Hope to see you again soon!");
+        Ui.setFlag();
+    }
+
+    /**
+     * Action taken when the command is list.
+     */
+    private void commandList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < TaskList.getList().size(); ++i) {
+            System.out.println(i + 1 + "." + TaskList.getList().get(i));
+        }
+    }
+
+    /**
+     * Action taken when the command is done.
+     */
+    private void commandDone() {
+        Storage storage = new Storage();
+        int num = Integer.parseInt(action[1]);
```

Comment:
This may be just me nitpicking haha but perhaps you can try to catch IndexOutOfBoundsException within this method.
After splitting the input (command) at " " in your Ui class and passing it into your Parser class as the array "action", it may not always be the case the array has a length more than 1. For instance, if I accidentally entered "done" instead of the correct "done 1" syntax, the array action would be of length 1 and calling action[1] may result in an IndexOutOfBoundsException which may unintentionally terminate the program. Hope that helps :)